### PR TITLE
test: write unit test for getDevicePathWithMountPath function

### LINKED ISSUE #688: write unit test for getDevicePathWithMountPath func
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/8f5975e141c40199d0155d77762e9e337c5a06e4/pkg/azuredisk/nodeserver.go#L561-L574

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by kassarl**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Adds a unit test for getDevicePathWithMountPath

**Which issue(s) this PR fixes**:

Fixes #688

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
thanks, could you fix the gofmt issue?

File: pkg/azuredisk/nodeserver_test.go [hunk 1]
```diff
@@ -893,3 +893,31 @@ func TestGetDevicePathWithLUN(t *testing.T) {
...
+		err = "executable file not found in $PATH"
+	}
+
+	tests := []struct {
+		desc        string
+		req         string
+		expectedErr error
+	}{
+		{
+			desc:        "Invalid device path",
```
Comment (andyzhangx) [inline pkg/azuredisk/nodeserver_test.go:912]: could you also add a valid test, e.g. `/sys`
```console
findmnt -o source --noheadings --mountpoint /sys
sysfs
```

Comment (kassarl) [reply pkg/azuredisk/nodeserver_test.go:912]: Is there a mount path equivalent for running on Mac OS? I have been looking but haven't had any luck

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

`MacOS Build & Unit Test / Build (pull_request)` test pass, so I think `/sys` works.

**Pull request merged**